azurerm/internal/services/machinelearning/resource_arm_machine_learning_workspace.go

```diff
@@ -0,0 +1,422 @@
+}
+
+func resourceArmMachineLearningWorkspaceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MachineLearning.WorkspacesClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.WorkspaceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing Machine Learning Workspace ID `%q`: %+v", d.Id(), err)
+	}
+
+	if _, err := client.Delete(ctx, id.ResourceGroup, id.Name); err != nil {
+		return fmt.Errorf("Error deleting Machine Learning Workspace %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	return nil
+}
+
+func validateStorageAccount(ctx context.Context, client *storage.AccountsClient, accountID string) error {
+	if accountID == "" {
+		return fmt.Errorf("Error validating Storage Account: Empty ID")
+	}
+
+	// TODO -- use parse function "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/parsers".ParseAccountID
+	// when issue https://github.com/Azure/azure-rest-api-specs/issues/8323 is addressed
+	id, err := parse.AccountIDCaseDiffSuppress(accountID)
+	if err != nil {
+		return fmt.Errorf("Error validating Storage Account: %+v", err)
+	}
+
+	account, err := client.GetProperties(ctx, id.ResourceGroup, id.Name, "")
+	if err != nil {
+		return fmt.Errorf("Error validating Storage Account %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+	if sku := account.Sku; sku != nil {
+		if sku.Tier == storage.Premium {
+			return fmt.Errorf("Error validating Storage Account %q (Resource Group %q): The associated Storage Account must not be Premium", id.Name, id.ResourceGroup)
+		}
+	}
+
+	return nil
+}
+
+func validateContainerRegistry(ctx context.Context, client *containerregistry.RegistriesClient, acrID *string) error {
+	if acrID == nil {
+		return nil
+	}
+
+	// TODO: use container registry's custom ID parse function when implemented
+	id, err := azure.ParseAzureResourceID(*acrID)
+	if err != nil {
+		return fmt.Errorf("Error validating Container Registry: %+v", err)
+	}
+
+	acrName := id.Path["registries"]
+	resourceGroup := id.ResourceGroup
+	acr, err := client.Get(ctx, resourceGroup, acrName)
+	if err != nil {
+		return fmt.Errorf("Error validating Container Registry %q (Resource Group %q): %+v", acrName, resourceGroup, err)
+	}
+	if acr.AdminUserEnabled == nil || !*acr.AdminUserEnabled {
+		return fmt.Errorf("Error validating Container Registry%q (Resource Group %q): The associated Container Registry must set `admin_enabled` to true", acrName, resourceGroup)
+	}
+
+	return nil
+}
+
+func expandArmMachineLearningWorkspaceIdentity(input []interface{}) *machinelearningservices.Identity {
+	if len(input) == 0 {
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
+
+	identityType := machinelearningservices.ResourceIdentityType(v["type"].(string))
+
+	identity := machinelearningservices.Identity{
+		Type: identityType,
+	}
+
+	return &identity
+}
```

Comment:
It was more if you did the complex `d.Get` you could have removed this entire function simplifying the code. minor so not going to block the PR on it

Reply:
I see, thank you!